providers/implementations/encode_decode/encode_key2text.c

```diff
@@ -0,0 +1,884 @@
+        void *key = ossl_prov_import_key(impl##_keymgmt_functions,      \
+                                         ctx, selection, params);       \
+        int ret;                                                        \
+                                                                        \
+        if (key == NULL)                                                \
+            return 0;                                                   \
+                                                                        \
+        ret = impl##_##kind##2text_encode_o(ctx, key, cout, cb, cbarg); \
+        ossl_prov_free_key(impl##_keymgmt_functions, key);              \
+        return ret;                                                     \
+    }                                                                   \
+    static int                                                          \
+    impl##_##kind##2text_encode_o(void *vctx, const void *key,          \
+                                  OSSL_CORE_BIO *cout,                  \
+                                  OSSL_PASSPHRASE_CALLBACK *cb,         \
+                                  void *cbarg)                          \
+    {                                                                   \
+        int selection = type##_##kind##_selection;                      \
+                                                                        \
+        return key2text_encode(vctx, key, selection, cout,              \
+                               type##_to_text, cb, cbarg);              \
+    }                                                                   \
+    const OSSL_DISPATCH impl##_##kind##_to_text_encoder_functions[] = { \
+        { OSSL_FUNC_ENCODER_NEWCTX,                                     \
+          (void (*)(void))key2text_newctx },                            \
+        { OSSL_FUNC_ENCODER_FREECTX,                                    \
+          (void (*)(void))key2text_freectx },                           \
+        { OSSL_FUNC_ENCODER_ENCODE_DATA,                                \
+          (void (*)(void))impl##_##kind##2text_encode_d },              \
+        { OSSL_FUNC_ENCODER_ENCODE_OBJECT,                              \
+          (void (*)(void))impl##_##kind##2text_encode_o },              \
+        { 0, NULL }                                                     \
+    }
+
+#define MAKE_TEXT_ENCODER(impl, type)                                   \
+    MAKE_TEXT_ENCODER_KIND(impl, param, type);                          \
+    MAKE_TEXT_ENCODER_KIND(impl, pub, type);                            \
+    MAKE_TEXT_ENCODER_KIND(impl, priv, type)
+
+#define MAKE_TEXT_ENCODER_NOPARAM(impl, type)                           \
+    MAKE_TEXT_ENCODER_KIND(impl, pub, type);                            \
+    MAKE_TEXT_ENCODER_KIND(impl, priv, type)
+
+#ifndef OPENSSL_NO_DH
+#define dh_text_params_bio i2d_DHparams_bio
```

Comment:
Are these defines a leftover??

Reply:
No, but they should be moved too their respective sectino

Reply:
they dont seem related to this macro..

Reply:
Oh wait, looks like you're right.  Ok, tossed